modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectStateFlowLocalServiceImpl.java

```diff
@@ -183,6 +183,48 @@ public ObjectStateFlow updateDefaultObjectStateFlow(
 		return null;
 	}
 
+	public void updateObjectStateFlowTransitions(
+			long userId, long objectStateFlowId, List<ObjectState> objectStates)
+		throws PortalException {
+
+		_objectStateTransitionLocalService.
+			deleteObjectStateFlowObjectStateTransitions(objectStateFlowId);
+
+		_objectStateLocalService.deleteObjectStateFlowObjectStates(
+			objectStateFlowId);
+
+		List<ObjectState> sourceObjectStates = TransformUtil.transform(
+			objectStates,
+			objectState -> {
+				ObjectState sourceObjectState =
+					_objectStateLocalService.addObjectState(
+						userId, objectState.getListTypeEntryId(),
+						objectStateFlowId);
+
+				sourceObjectState.setObjectStateTransitions(
+					objectState.getObjectStateTransitions());
+
+				return sourceObjectState;
+			});
+
+		for (ObjectState sourceObjectState : sourceObjectStates) {
+			for (ObjectStateTransition objectStateTransition :
+					sourceObjectState.getObjectStateTransitions()) {
+
+				ObjectState targetObjectState =
+					_objectStateLocalService.getObjectStateFlowObjectState(
+						objectStateTransition.
+							getTargetObjectStateListTypeEntryId(),
+						objectStateFlowId);
+
+				_objectStateTransitionLocalService.addObjectStateTransition(
+					userId, objectStateFlowId,
+					sourceObjectState.getObjectStateId(),
+					targetObjectState.getObjectStateId());
+			}
+		}
```

Comment:
@G-Sa this code is the same from [here](https://github.com/liferay-objects/liferay-portal/blob/3de804b5974a700730551b5235514725085686c0/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectStateFlowLocalServiceImpl.java#L87-L116). 

Please move it to a common method to avoid duplicate code.